alacritty/src/input.rs

```diff
@@ -366,23 +366,66 @@ impl<'a, T: EventListener, A: ActionContext<T>> Processor<'a, T, A> {
         }
     }
 
-    pub fn normal_mouse_report(&mut self, button: u8) {
+    pub fn utf8_mouse_report(&mut self, button: u8) {
         let (line, column) = (self.ctx.mouse().line, self.ctx.mouse().column);
 
-        if line < Line(223) && column < Column(223) {
-            let msg = vec![
+        if line < Line(2015) && column < Column(2015) {
+            let mouse_pos_encode = |pos: usize| -> (u8, u8) {
+                let pos = 32 + 1 + pos;
+                let first = 0xC0 + (pos >> 6);
+                let second =  0x80 + (pos & 0x3F);
+                (first as u8, second as u8)
+            };
+
+            let mut msg = vec![
                 b'\x1b',
                 b'[',
                 b'M',
                 32 + button,
-                32 + 1 + column.0 as u8,
-                32 + 1 + line.0 as u8,
             ];
 
+            if column < Column(95) {
+                msg.push(32 + 1 + column.0 as u8);
+            } else {
+                let encoded_col = mouse_pos_encode(column.0);
+                msg.push(encoded_col.0);
+                msg.push(encoded_col.1);
+            }
+
+            if line < Line(95) {
+                msg.push(32 + 1 + line.0 as u8);
+            } else {
+                let encoded_line = mouse_pos_encode(line.0);
+                msg.push(encoded_line.0);
+                msg.push(encoded_line.1);
+            }
```

Comment:
Disregarding everything I've said about moving the call to the `utf8_mouse_report` into the `mouse_report` function, couldn't we just merge the normal and utf8 code together like this?

```rust
// I'm not actually sure if '>=' is correct here, that's what we're doing right now
// following the xterm docs though, they say the last position is 223/2015, making it sound inclusive
if (utf8 && line >= 2015) || (!utf8 && line >= 223) { return }

if utf8 && line < Line(95) {
    // ...
} else if utf8 {
    // ...
}
```

Reply:
In code they're sending `zero` for 2015 and 233. With doesn't make sense to me, but maybe this is how it's supposed to work.

Reply:
I'm not sure how that is related to this comment?